src/commands/schemaUpload.ts

```diff
@@ -179,11 +180,12 @@ async function uploadSchema(
     // Get the schemas view provider to refresh the view on the right registry.
     // (The resource manager data for that registry will be updated with the new schema
     //  via updateRegistryCacheAndFindNewSchema() which has already resolved.)
-    currentSchemaRegistryChanged.fire(registry);
+    if (schemaViewProvider.schemaRegistry?.id !== registry.id) {
+      currentSchemaRegistryChanged.fire(registry);
```

Comment:
Was this the cause of the double-refreshing we saw yesterday? 👀

Reply:
Maybe. We saw a lot of things yesterday.